lib/system/web/src/keyboard_engine.rs

```diff
@@ -0,0 +1,122 @@
+
+        KeyboardEngine::init(target, Rc::clone(&bindings), Rc::clone(&state));
+
+        Ok(Self {
+            bindings,
+            state,
+            target: target.clone()
+        })
+    }
+
+    pub fn from_tag_name(html_tag: &str) -> Result<Self> {
+        let element = document()?
+            .get_elements_by_tag_name(html_tag)
+            .item(0).ok_or_else(|| Error::missing("there is no such element"));
+
+        KeyboardEngine::new(&element.unwrap())
+    }
+
+    fn init (
+        target:   &web_sys::EventTarget,
+        bindings: Rc<RefCell<Bindings>>,
+        state:    Rc<RefCell<BTreeSet<u32>>>
+    ) {
+        {
+            let state = Rc::clone(&state);
+            let bindings = Rc::clone(&bindings);
+            let keydown_callback = Box::new(move |event: web_sys::KeyboardEvent| {
+                if event.repeat() {
+                    return;
+                }
+
+                let key_code = event.key_code();
+                state.borrow_mut().insert(key_code);
+
+                let key = state.borrow();
+                if let Some(vec) = bindings.borrow_mut().get_mut(&key) {
+                    vec.iter_mut().for_each(|func| func());
+                }
+            });
+            let keydown_callback = Closure::wrap(
+                keydown_callback as Box<dyn FnMut(_)>);
+            target.add_event_listener_with_callback(
+                "keydown", keydown_callback.as_ref().unchecked_ref())
+                .unwrap();
+            keydown_callback.forget();
+        }
+
+        {
+            let keyup_callback = Box::new(move |event: KeyboardEvent| {
+                let key_code = event.key_code();
+                state.borrow_mut().remove(&key_code);
+
+                let key = state.borrow();
+                if let Some(vec) = bindings.borrow_mut().get_mut(&key) {
+                    vec.iter_mut().for_each(|func| func());
+                }
+            });
+            let keyup_callback = Closure::wrap(
+                keyup_callback as Box<dyn FnMut(_)>);
+            target.add_event_listener_with_callback(
+                "keyup", keyup_callback.as_ref().unchecked_ref())
+                .unwrap();
+            keyup_callback.forget();
+        }
+    }
+
+    /// Capture key combination
+    ///
+    /// Uses u32 as key code from WebIDL
+    /// https://github.com/rustwasm/wasm-bindgen/blob/913fdbc3daff65952b5678a34b98e07d4e6e4fbb/crates/web-sys/webidls/enabled/KeyEvent.webidl
+    pub fn capture(
+        & self,
+        combination: Vec<u32>,
+        callback:    Box<dyn FnMut()>
+    ) {
+        let key = combination.iter().copied().collect::<BTreeSet<u32>>();
+        let mut bindings = self.bindings.borrow_mut();
+        match bindings.get_mut(&key) {
+            Some(callbacks) => {
+                callbacks.push(callback);
+            }
+            None => {
+                bindings.insert(key, vec![callback]);
+            }
+        }
```

Comment:
This logic is a responsibility of some kind of callback manager, not capturing function!

Reply:
Why not? I'm capturing here the key press, and callback management is a part of implementation